packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -964,13 +964,18 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
 ///
 /// ### Nested Scaffolds
 ///
-/// The Scaffold was designed to be the single top level container for
-/// a [MaterialApp] and it's typically not necessary to nest
-/// scaffolds. For example in a tabbed UI, where the
-/// [bottomNavigationBar] is a [TabBar] and the body is a
-/// [TabBarView], you might be tempted to make each tab bar view a
-/// scaffold with a differently titled AppBar. It would be better to add a
-/// listener to the [TabController] that updates the AppBar.
+/// The Scaffold was designed to be a single top level container for
+/// most of the content in a route(although, not necessarly the first widget).
+/// It is recommended that you use Scaffold once per route(adhering to
+/// material guidelines) for most of the simple cases, unless you really
+/// need to nest them to achieve the functionality you need.
+///
+/// You'll find that it's typically not necessary to nest scaffolds. For
+/// example, in a tabbed UI, where the [bottomNavigationBar] is a [TabBar]
+/// and the body is a [TabBarView], you might be tempted to make each tab bar
+/// view a scaffold with a differently titled AppBar. Rather, it would be
+/// better to add a listener to the [TabController] that updates the
+/// AppBar.
```

Comment:
A few comments and my suggestion:
- "First widget" is vague here -- Did you mean the top-level widget?
- I would avoid using words like "recommend" or "refrain", since a user has the liberty to decide what they want to do with their apps. We should opt to describe what the widget does and what outcome it'll help the developer achieve.

```suggestion
/// The Scaffold is designed to be a top level container for
/// a [MaterialApp]. This means that adding a Scaffold
/// to each route on a Material app will provide the app with
/// Material's basic visual layout structure.
///
/// It is typically not necessary to nest Scaffolds. For example, in a 
/// tabbed UI, where the [bottomNavigationBar] is a [TabBar]
/// and the body is a [TabBarView], you might be tempted to make each tab bar
/// view a scaffold with a differently titled AppBar. Rather, it would be
/// better to add a listener to the [TabController] that updates the
/// AppBar.
```

Reply:
Sure. These make a lot more sense than before, and also subtly bring across the fact that we can add a scaffold to each route. Committing these changes